rbac/api/cross_access/model.py

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Models to store cross account access request."""
+from uuid import uuid4
+
+from django.db import models
+from django.utils import timezone
+from management.rbac_fields import AutoDateTimeField
+
+STATUS_LIST = ["requested", "cancelled", "approved", "denied"]
```

Comment:
Would we expectation be that the "expired" state be implicit based on the start/end date, and not necessarily the status state in the db?

Reply:
I guess that would incur the effort of calculating whether or not a given request has expired whenever a request is made. As long as there are a small number of requests at a given time, I don't expect it to be a performance burden. In the integration tests we'll probably want to figure out how to create a given request and force it to expire. On that note, will there be any validation of the start/end date at the time the request is created?

Reply:
Maybe just a start/end date off now()? I was thinking about the implicit state as well. I suppose that would be a possibility to have an internal check on the state when a request object is requested, to update it before the response if it's expired?

Reply:
yea, I was thinking about the "expired" state as implicit based on the end date. That does require extra calculation. 
It would be useful to make a scheduler to expire it (update the db status) on end date then, that way it won't introduce extra calculation when listing request objects.